deployment/common/changeset/set_config_mcms.go

```diff
@@ -0,0 +1,206 @@
+}
+type ProposalConfig struct {
+	MinDelay time.Duration // delay for timelock worker to execute the transfers.
+}
+
+type SetConfigParams struct {
+	ConfigsPerChain map[uint64]ConfigPerRole
+	ProposalConfig  *ProposalConfig
+}
+
+var _ deployment.ChangeSet[SetConfigParams] = SetConfigMCMS
+
+// Validate checks that the SetConfigParams is valid
+func (cfg SetConfigParams) Validate(e deployment.Environment, selectors []uint64) error {
+	// configs should have at least one chain
+	state, err := MaybeLoadMCMSWithTimelockState(e, selectors)
+	if err != nil {
+		return err
+	}
+	if len(cfg.ConfigsPerChain) == 0 {
+		return errors.New("no chain configs provided")
+	}
+	for chainSelector, c := range cfg.ConfigsPerChain {
+		family, err := chain_selectors.GetSelectorFamily(chainSelector)
+		if err != nil {
+			return err
+		}
+		if family != chain_selectors.FamilyEVM {
+			return fmt.Errorf("chain selector: %d is not an ethereum chain", chainSelector)
+		}
+		_, ok := e.Chains[chainSelector]
+		if !ok {
+			return fmt.Errorf("chain selector: %d not found in environment", chainSelector)
+		}
+		_, ok = state[chainSelector]
+		if !ok {
+			return fmt.Errorf("chain selector: %d not found for MCMS state", chainSelector)
+		}
+		if err := c.Proposer.Validate(); err != nil {
+			return err
+		}
+		if err := c.Canceller.Validate(); err != nil {
+			return err
+		}
+		if err := c.Bypasser.Validate(); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+// setConfigOrTxData executes set config tx or gets the tx data for the MCMS proposal
+func setConfigOrTxData(chain deployment.Chain, cfg config.Config, contract *gethwrappers.ManyChainMultiSig, useMCMS bool) (*types.Transaction, error) {
+	groupQuorums, groupParents, signerAddresses, signerGroups := cfg.ExtractSetConfigInputs()
+	opts := deployment.SimTransactOpts()
+	if !useMCMS {
+		opts = chain.DeployerKey
+	}
+	tx, err := contract.SetConfig(opts, signerAddresses, signerGroups, groupQuorums, groupParents, false)
+	if err != nil {
+		return nil, err
+	}
+	if !useMCMS {
+		_, err = deployment.ConfirmIfNoError(chain, tx, err)
+		if err != nil {
+			return nil, err
+		}
+	}
+	return tx, nil
+}
+
+type setConfigTxs struct {
+	proposerTx  *types.Transaction
+	cancellerTx *types.Transaction
+	bypasserTx  *types.Transaction
+}
+
+// setConfigPerRole sets the configuration for each of the MCMS contract roles on the mcmsState.
+func setConfigPerRole(chain deployment.Chain, cfg ConfigPerRole, mcmsState *MCMSWithTimelockState, useMCMS bool) (setConfigTxs, error) {
+	// Proposer set config
+	proposerTx, err := setConfigOrTxData(chain, cfg.Proposer, mcmsState.ProposerMcm, useMCMS)
+	if err != nil {
+		return setConfigTxs{}, err
+	}
+	// Canceller set config
+	cancellerTx, err := setConfigOrTxData(chain, cfg.Canceller, mcmsState.CancellerMcm, useMCMS)
+	if err != nil {
+		return setConfigTxs{}, err
+	}
+	// Bypasser set config
+	bypasserTx, err := setConfigOrTxData(chain, cfg.Bypasser, mcmsState.BypasserMcm, useMCMS)
+
```

Comment:
nitpick: sorry for being super nitpick, just mainly format consistency, we have a new line before the `err` check where as the ones above do not, so we should remove the new line here.

Reply:
added here! https://github.com/smartcontractkit/chainlink/pull/15684/commits/d8cc495842e298e8a6b212fdb5c3bea4bd0929b5